custom_components/detailed_hello_world/config_flow.py

```diff
@@ -0,0 +1,113 @@
+    if len(data["host"]) < 3:
+        raise InvalidHost
+
+    hub = Hub(hass, data["host"])
+    # The dummy hub provides a `test_connection` method to ensure it's working
+    # as expected
+    result = await hub.test_connection()
+    if not result:
+        # If there is an error, raise an exception to notify HA that there was a
+        # problem. The UI will also show there was a problem
+        raise CannotConnect
+
+    # If your PyPI package is not built with async, pass your methods
+    # to the executor:
+    # await hass.async_add_executor_job(
+    #     your_validate_func, data["username"], data["password"]
+    # )
+
+    # If you cannot connect:
+    # throw CannotConnect
+    # If the authentication is wrong:
+    # InvalidAuth
+
+    # Return info that you want to store in the config entry.
+    # "Title" is what is displayed to the user for this hub device
+    # It is stored internally in HA as part of the device config.
+    # See `async_step_user` below for how this is used
+    return {"title": data["host"]}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Hello World."""
+
+    VERSION = 1
+    # Pick one of the available connection classes in homeassistant/config_entries.py
+    # This tells HA if it should be asking for updates, or it'll be notified of updaes
+    # automatically. This example uses PUSH, as the dummy hub will notify HA of
+    # changes.
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        # This goes through the steps to take the user through the setup process.
+        # Using this it is possible to update the UI and prompt for additional
+        # information. This example provides a single form (built from `DATA_SCHEMA`),
+        # and when that has some validated input, it calls `async_create_entry` to
+        # actually create the HA config entry. Note the "title" value is returned by
+        # `validate_input` above.
+        errors = {}
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+
+                return self.async_create_entry(title=info["title"], data=user_input)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidHost:
+                # The error string is set here, and should be translated.
+                # This example does not currently cover translations, see the
+                # comments on `DATA_SCHEMEA` for further details.
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+        # If there is no user input or there were errors, show the form again, including any errors that were found with the input.
```

Comment:
I think that a bunch of these comments should also be put back into our scaffolding templates https://github.com/home-assistant/core/blob/dev/script/scaffold/templates/config_flow/integration/config_flow.py#L81